src/backups2datalad/adataset.py

```diff
@@ -105,6 +108,60 @@ async def ensure_installed(
         log.debug("Dataset for %s created", desc)
         return True
 
+    async def ensure_dandi_provider(self, api_url: str) -> None:
+        prov_cfg = Path(".datalad", "providers", "dandi.cfg")
+        provider_file = self.pathobj / prov_cfg
+        if not provider_file.exists():
+            url_re = re.escape(api_url)
+            url_re = re.sub(r"^https?:", "https?:", url_re)
```

Comment:
I am confused -- what are you trying to achieve here? did you debug/see on how it looks in the end?

Reply:
The goal is to produce a line of the form `url_re = https?://api\-staging\.dandiarchive\.org/api/.*`, like in your script, where the periods in the URL have been escaped and the `s` in `https:` is optional.